Firestore/core/test/unit/api/aggregate_query_test.cc

```diff
@@ -0,0 +1,171 @@
+    return aggregate_query.query_;
+  }
+
+  static const std::vector<AggregateField>& GetAggregates(
+      const AggregateQuery& aggregate_query) {
+    return aggregate_query.aggregates_;
+  }
+};
+
+namespace {
+
+// Assert that the Get member function calls GetAggregate member function
+// and that the result from GetAggregate is processed
+// appropriately
+TEST(AggregateQuery, GetCallsGetAggregateOk) {
+  // Test aggregate field result
+  google_firestore_v1_AggregationResult_AggregateFieldsEntry
+      aggregate_fields_entry[1];
+  aggregate_fields_entry[0].key = nanopb::ByteString("aggregate_0").release();
+  aggregate_fields_entry[0].value.which_value_type =
+      google_firestore_v1_Value_integer_value_tag;
+  aggregate_fields_entry[0].value.integer_value = 10;
+
+  // Test alias map
+  absl::flat_hash_map<std::string, std::string> alias_map;
+  alias_map["aggregate_0"] = "count";
+
+  // Test ObjectValue result
+  ObjectValue object_value_result = ObjectValue::FromAggregateFieldsEntry(
+      aggregate_fields_entry, 1, alias_map);
+
+  // Create an AggregateQuery with mocked GetAggregate function that
+  // invokes the callback with the test results from above
+  AggregateField count_aggregate_field(AggregateField::OpKind::Count,
+                                       AggregateAlias("count"));
+  std::vector<AggregateField> aggregates{count_aggregate_field};
+  MockAggregateQuery mock_aggregate_query({}, std::move(aggregates));
+  EXPECT_CALL(mock_aggregate_query, GetAggregate)
+      .Times(1)
+      .WillOnce(Invoke([object_value_result = std::move(object_value_result)](
+                           AggregateQueryCallback&& callback) {
+        callback(object_value_result);
+      }));
+
+  // Call the Get function, which is the function under test
+  int callback_count = 0;
+  mock_aggregate_query.Get(
+      [&callback_count](const StatusOr<int64_t>& result) mutable {
+        callback_count++;
+        ASSERT_EQ(result.ok(), true);
+        EXPECT_EQ(result.ValueOrDie(), 10);
+      });
+
+  // Assert the callback was invoked
+  EXPECT_EQ(callback_count, 1);
+}
+
+// Assert that the Get member function calls GetAggregate member function
+// and that an error result from GetAggregate is processed
+// appropriately
+TEST(AggregateQuery, GetCallsGetAggregateError) {
+  // Error result
+  Status error_result = Status(Error::kErrorInternal, "foo");
+
+  // Create an AggregateQuery with mocked GetAggregate function that
+  // invokes the callback with the error status from above
+  Query query;
+  AggregateField count_aggregate_field(AggregateField::OpKind::Count,
+                                       AggregateAlias("count"));
+  std::vector<AggregateField> aggregates{count_aggregate_field};
+  MockAggregateQuery mock_aggregate_query(query, std::move(aggregates));
+  EXPECT_CALL(mock_aggregate_query, GetAggregate)
+      .Times(1)
+      .WillOnce(Invoke(
+          [error_result = std::move(error_result)](
```

Comment:
Optional: Rather than capturing `error_result`, why not just create the object in the lambda's body? IMO it's more readable that way because the logic is simpler and you don't have to scroll up 22 lines to find the definition of `error_result`.

Reply:
I understand your point of view but that would break structure of act-arrange-assert. go/unit-test-practices#structure

Reply:
How so? Both the definition of `error_result` and `EXPECT_CALL` are part of the "arrange" block, no?

Reply:
(for the record, I fully support the arrange-act-assert style of test structure)

Reply:
I guess that's true that expect is part of arrange, but the pattern of setting up the test doubles at the top of the test is used in this file. Changing that here would be inconsistent. Changing that everywhere would make the code hard to read.